src/tribler/core/utilities/process_locker.py

```diff
@@ -0,0 +1,362 @@
+    return json.dumps(value)
+
+
+def from_json(value) -> Optional[dict]:
+    if value is None:
+        return None
+    return json.loads(value)
+
+
+class ProcessInfo:
+    def __init__(self, pid: int, kind: ProcessKind, app_version: str, started_at: int,
+                 rowid: Optional[int] = None, creator_pid: Optional[int] = None, active: int = 0, canceled: int = 0,
+                 row_version: int = 0, api_port: Optional[int] = None, finished_at: Optional[int] = None,
+                 exit_code: Optional[int] = None, error_msg: Optional[str] = None, error_info: Optional[dict] = None,
+                 shutdown_request_pid: Optional[int] = None, shutdown_requested_at: Optional[int] = None,
+                 other_params: Optional[dict] = None):
+        self.rowid = rowid
+        self.row_version = row_version
+        self.pid = pid
+        self.kind = kind
+        self.active = active
+        self.canceled = canceled
+        self.app_version = app_version
+        self.started_at = started_at
+        self.creator_pid = creator_pid
+        self.api_port = api_port
+        self.finished_at = finished_at
+        self.exit_code = exit_code
+        self.error_msg = error_msg
+        self.error_info = error_info
+        self.shutdown_request_pid = shutdown_request_pid
+        self.shutdown_requested_at = shutdown_requested_at
+        self.other_params = other_params
+
+    @classmethod
+    def from_row(cls, row: tuple) -> ProcessInfo:
+        rowid, row_version, pid, kind, active, canceled, app_version, started_at, creator_pid, api_port, \
+            shutdown_request_pid, shutdown_requested_at, finished_at, exit_code, error_msg, error_info, \
+            other_params = row
+
+        kind = ProcessKind(kind)
+
+        return ProcessInfo(rowid=rowid, row_version=row_version, pid=pid, kind=kind, active=active, canceled=canceled,
+                           app_version=app_version, started_at=started_at, creator_pid=creator_pid,
+                           api_port=api_port, shutdown_request_pid=shutdown_request_pid,
+                           shutdown_requested_at=shutdown_requested_at, finished_at=finished_at,
+                           exit_code=exit_code, error_msg=error_msg, error_info=from_json(error_info),
+                           other_params=from_json(other_params))
+
+    def to_dict(self) -> dict:
+        d = dict(rowid=self.rowid, pid=self.pid, kind=self.kind.value, active=self.active, canceled=self.canceled,
+                 app_version=self.app_version, started_at=self.started_at, creator_pid=self.creator_pid,
+                 api_port=self.api_port, finished_at=self.finished_at, exit_code=self.exit_code,
+                 error_msg=self.error_msg, error_info=self.error_info, shutdown_request_pid=self.shutdown_request_pid,
+                 shutdown_request_at=self.shutdown_requested_at, other_params=self.other_params)
+        return {key: value for key, value in d.items() if value is not None}
```

Comment:
NIT: maybe it is easier to derive the class from the `dataclass` to omit this conversion?

Reply:
I considered this option, but I think it is more like a "normal" class now and contains enough logic to not be derived from the `dataclass`.